### PR TITLE
Session refactor and accessibility service

### PR DESCRIPTION
**Pull request by aaronhuggins**
Removes all changes to gradle version, additional project files, etc.

### CODE REVIEW

File: js/CobrowseIO.js [hunk 1]
```diff
@@ -4,6 +4,112 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
 
 const emitter = new NativeEventEmitter(CobrowseIONative)
 
+class Session {
```
Comment (headlessme) [inline js/CobrowseIO.js:7]: Any reason this is in the CobrowseIO main file? Can you move to Session.js?

File: js/CobrowseIO.js [hunk 2]
```diff
@@ -4,6 +4,112 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
...
+  setFullDevice (state) {
+    return CobrowseIONative.updateSession({ full_device: state })
+  }
+
+  setRemoteControl (state) {
+    return CobrowseIONative.updateSession({ remote_control: state })
+  }
+}
+
+class CobrowseAccessibilityService {
```
Comment (headlessme) [inline js/CobrowseIO.js:103]: Move to CobrowseAccessibilityService.js if you can

File: js/CobrowseIO.js [hunk 3]
```diff
@@ -4,6 +4,112 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
...
+  }
+
+  get agent () {
+    return this._session.agent
+  }
+
+  activate () {
+    return CobrowseIONative.activateSession().then((session) => {
+      this._session = session
+    })
```
Comment (headlessme) [inline js/CobrowseIO.js:67]: Is the then() actually required here? The session activation change should come back through the updated event as well

File: js/CobrowseIO.js [hunk 4]
```diff
@@ -4,6 +4,112 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
...
+  }
+
+  end () {
+    return CobrowseIONative.endSession()
+  }
+
+  hasAgent () {
+    return !!(this._session.agent)
+  }
```
Comment (headlessme) [inline js/CobrowseIO.js:76]: I don't think we have this on the agent API? I know it's on the Android API but probably not actually needed as we have the agent getter

File: js/CobrowseIO.js [hunk 5]
```diff
@@ -4,6 +4,112 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
...
+
+  _update (session, onEnd) {
+    if (this._session.id === undefined || (session && this._session.id === session.id)) {
+      this._session = session
+      if (typeof onEnd === 'function') onEnd()
+    }
+  }
+
+  _listen () {
+    const requests = emitter.addListener(CobrowseIONative.SESSION_REQUESTED, (session) => this._update(session))
```
Comment (headlessme) [inline js/CobrowseIO.js:21]: I don't think the session object actually needs to know about the requested events (they're a oddity of the RN APIs).

File: js/CobrowseIO.js (hunk 4 above)
Comment (aaronhuggins) [reply js/CobrowseIO.js:76]: It's on the web sdk and the android sdk. I felt it was appropriate to also expose on the react-native sdk for feature-parity.

File: js/CobrowseIO.js (hunk 5 above)
Comment (aaronhuggins) [reply js/CobrowseIO.js:21]: I see what you mean now. I'll remove it.

File: js/CobrowseIO.js (hunk 4 above)
Comment (aaronhuggins) [reply js/CobrowseIO.js:76]: Oh, actually, I misinterpreted the web sdk. What I thought of as "hasAgent" is just a method called "agent" which returns the agent object or the value `false`.

File: js/CobrowseIO.js (hunk 3 above)
Comment (aaronhuggins) [reply js/CobrowseIO.js]: This does guarantee that if `activate` is called, it comes back void. I'll remove the variable assignment, but leave the void function so the promise returns reliably. I don't want the class to leak the wrapped session object.

**Pull request merged**